python/lsst/daf/butler/cli/butler.py

```diff
@@ -79,36 +90,80 @@ def _importPlugin(pluginName):
             return None
 
     @staticmethod
-    def _getLocalCommand(commandName):
-        """Search the commands provided by this package and return the command
-        if found.
+    def _mergeCommandLists(a, b):
+        """Combine two dicts whose keys are strings (command name) and values
+        are list of string (the package(s) that provide the named command).
 
         Parameters
         ----------
-        commandName : string
-            The name of the command to search for.
+        a : dict of string to list of string
+            key is command name, value is packages that provide the command
+        b : (same as a)
 
         Returns
         -------
-        Command object
-            A Click command that can be executed.
+        dict of string to list of string
+            returns a extended with b
         """
-        try:
-            return getattr(butlerCommands, commandName)
-        except AttributeError:
-            pass
-        return None
+        for key, val in b.items():
+            try:
+                a[key].extend(val)
```

Comment:
I'm not a huge fan of using exceptions for expected behavior. Since this is internal API I think it might be cleaner to set up `commands` below as a `defaultdict`. Then you won't need the try at all. (since you test it as if it is public API that might lead to too much work having to ensure that the first dict is a defaultdict with the right default).